### ISSUE TITLE
New scaffolded projects contain phantom dependencies and do not compile with pnpm 

### ISSUE THREAD

**Issue by stevebeauge** [SharePoint/sp-dev-docs#8945]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

SPFX 1.17.1 (but bug present for ages)
Node 16.20.1

### Describe the bug / error

Projects that are scaffolded using the official `@microsoft/generator-sharepoint` generator contains phantom dependencies.

Phantom dependencies are [well explained by Microsoft in their rushstack doc](https://rushjs.io/pages/advanced/phantom_deps).

Using `npm` won't cause visible issue (which is like putting dust under the carpet). But the issue is revealed when using more robust package managers like `pnpm`.

## `@fluentui/react`

1st missing dependency is `@fluentui/react` which is missing and prevent building scss files:

````
Error - sass - src\webparts\myWebPart\components\MyWebPart.module.scss(1,9): error Error: Can't find stylesheet to import.
  ╷
1 │ @import '~@fluentui/react/dist/sass/References.scss';
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  src\webparts\myWebPart\components\MyWebPart.module.scss 1:9  root stylesheet
[22:32:29] Error - 'sass' sub task errored after 95 ms
 Can't find stylesheet to import.
  ╷
1 │ @import '~@fluentui/react/dist/sass/References.scss';
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  src\webparts\myWebPart\components\MyWebPart.module.scss 1:9  root stylesheet
````

Solved by adding `@fluentui/react@8.109.1` dependency explicitly.

## '@microsoft/sp-component-base'

2nd missing dependency is '@microsoft/sp-component-base'.

````
Error - [tsc] src/webparts/myWebPart/MyWebPartWebPart.ts(9,32): error TS2307: Cannot find module '@microsoft/sp-component-base' or its corresponding type declarations.
Error - 'tsc' sub task errored after 4.58 s

````

Same fix: add `@microsoft/sp-component-base@1.17.1` explicitly to solve the issue.

## Proposed solution

Obviously, the solution should be to include these two dependencies explicitly in the template, and any future explicit usage of a dependency. MS should also include `yarn` and `pnpm` package managers in their tests since their adoption is growing very high (npm has very low performance and have some issues regarding dependency graph - check Rustask for explanations)

### Steps to reproduce

Install pnpm v8 and node 16

1. bootstrap a new SPFX project. Choose to create a `webpart` / `react` project.
 ````powershell
 pnpm --package yo@latest --package @microsoft/generator-sharepoint@1.17.1 dlx yo @microsoft/sharepoint@latest --package-manager pnpm --component-type webpart --framework react --solution-name "repro-spfx" --component-name MyWebPart
 ````
2. install dependencies
  ````powershell
  cd repro-spfx
  pnpm install
  ````
3. Try to build
  ````powershell
  pnpm build
  ````
4. ❌ Does not compiles (missing `@fluentui/react`)
5. Add the missing dependency and try to build again
  ````powershell
  pnpm add @fluentui/react@8.109.1
  pnpm build
  ````
6. ❌ Does not compiles (missing `@microsoft/sp-component-base`)
7. Add the missing dependency and try to build
  ````powershell
  pnpm add @microsoft/sp-component-base@1.17.1
  pnpm build
  ````
8. ✅ This time, every compiles properly

### Expected behavior

Dependencies should be setup properly, in order to avoid phantom dependencies (which is a good practice I guess), and it should work out of the box with the major package managers (npm, yarn and pnpm).

**Comment by AJIXuMuK** [comment]
dupe of #8939

**Comment by AJIXuMuK** [comment]
fixed in 1.17.2
